src/main/java/org/cactoos/iterator/Filtered.java

```diff
@@ -73,8 +73,8 @@
      * @param fnc Predicate
      * @param src Source iterable
      */
-    public Filtered(final Func<X, Boolean> fnc, final Iterator<X> src) {
-        this.iterator = src;
+    public Filtered(final Func<X, Boolean> fnc, final Iterable<X> src) {
+        this.iterator = src.iterator();
```

Comment:
@mazdack I believe that for all of the `org.cactoos.iterator` class like this one, we shouldn't remove the `Iterator` from the constructors since they are! Please put them all back then